src/main/java/org/broadinstitute/hellbender/utils/diffengine/BAMDiffableReader.java

```diff
@@ -0,0 +1,82 @@
+package org.broadinstitute.hellbender.utils.diffengine;
+
+import htsjdk.samtools.*;
+import htsjdk.samtools.util.BlockCompressedInputStream;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+
+import java.io.*;
+import java.util.Arrays;
+
+/**
+ * Class implementing diffnode reader for BAM/SAM/CRAM files.
```

Comment:
Should not claim to work with SAM and CRAM, since it explicitly checks for the BAM-specific magic value in `canRead()`, and does not set up its `SAMReader` properly for CRAM.